src/sentry/region_to_control/producer.py

```diff
@@ -1,28 +1,24 @@
-from typing import Any, Dict, Optional
+import dataclasses
+from typing import Optional
 
 from django.conf import settings
 
-from sentry.models import User, UserIP
+from sentry.models import UserIP
+from sentry.region_to_control.messages import UserIpEvent
 from sentry.silo import SiloMode
 from sentry.utils import json, kafka_config
 from sentry.utils.pubsub import KafkaPublisher
 
 
-def produce_user_ip(user: User, ip_address: str, values: Dict[str, Any]):
+def produce_user_ip(event: UserIpEvent):
+    values = dataclasses.asdict(event)
```

Comment:
@markstory @mikejihbe 
So this is where I landed -- in this way, generally the difference between control and regional behavior is non existent because the message matches the schema being posted.  However, and to your point mike, the consumer side needs to be more careful to be able to support 'out of date messages' during rollouts and deploys.  For this reason consumer code path should not be converged with the producer code path in case any repackaging or defaulting is necessary to upgrade old kafka message versions into the new schema.

The problem of general schema evolution can be solved by other means, such as again via the approach Avro takes where 'schema upgrade' metadata becomes _part of the messaging protocol_.  

But, that said, the simplest solution, what I'm doing here, is assuming developers write robust upgrade code.  To facilitate this, I intend to create tests that use fixed payloads in the consumer tests and advice in the comments that developers duplicate and append new tests for new payload schemas, rather than update existing tests.